packages/dendron-next-server/hooks/useGraphElements.tsx

```diff
@@ -0,0 +1,186 @@
+import {
+  DVault,
+  NotePropsDict,
+  NoteUtils,
+  SchemaModuleDict,
+} from "@dendronhq/common-all";
+import { engineSlice } from "@dendronhq/common-frontend";
+import { EdgeDefinition, NodeDefinition } from "cytoscape";
+import _ from "lodash";
+import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
+import { GraphEdges, GraphElements, GraphNodes } from "../lib/graph";
+
+const getNoteGraphElements = (
+  notes: NotePropsDict,
+  wsRoot: string
+): GraphElements => {
+  // ADD NODES
+  const nodes = Object.values(notes).map((note) => ({
+    data: { id: note.id, label: note.title, group: "nodes" },
+  }));
+
+  // ADD EDGES
+  const edges: GraphEdges = {
+    hierarchy: [],
+    links: [],
+  };
+
+  Object.values(notes).forEach((note) => {
+    edges.hierarchy.push(
+      ...note.children.map((child) => ({
+        data: {
+          group: "edges",
+          id: `${notes.id}_${child}`,
+          source: note.id,
+          target: child,
+        },
+        classes: "hierarchy",
+      }))
+    );
+
+    const linkConnections: EdgeDefinition[] = [];
+
+    // Find and add linked notes
+    note.links.forEach((link) => {
+      if (link.to && note.id) {
+        const to = NoteUtils.getNoteByFnameV5({
+          fname: link.to!.fname as string,
+          vault: note.vault,
+          notes: notes,
+          wsRoot,
+        });
+
+        if (!to) return;
+        linkConnections.push({
+          data: {
+            group: "edges",
+            id: `${note.id}_${to.id}`,
+            source: note.id,
+            target: to.id,
+          },
+          classes: "links",
+        });
+      }
+    });
+
+    edges.links.push(...linkConnections);
+  });
+
+  return {
+    nodes,
+    edges,
+  };
+};
+
+const getSchemaGraphElements = (
+  schemas: SchemaModuleDict,
+  vaults: DVault[] | undefined
+): GraphElements => {
+  const schemaArray = Object.values(schemas);
+
+  const nodes: any[] = [];
+  const edges: GraphEdges = {
+    hierarchy: [],
+  };
+
+  if (_.isUndefined(vaults)) return { nodes, edges };
+
+  vaults.map((vault) => {
+    const VAULT_ID = `${vault.name}`;
```

Comment:
there's a chance that a vault does not have a name in which case we default to fsPath. should use [VaultUtils.getName](https://github.com/dendronhq/dendron/blob/feature-graph/packages/common-all/src/vault.ts#L8:L8)